site/topics/if-else/if-else.rst

```diff
@@ -1,45 +1,89 @@
-**************
-Logic, if/else
-**************
+*******
+If/Else
+*******
 
-Conditionals
-============
+.. admonition:: Activity
 
-* We're pretty good at telling python *exactly what to do* now.
-* But there is no room for choice. It's just: *do these steps in this order*
-* That doesn't give us much flexibility (or computational power)
+    Using only what we have learned so far, write a function ``smush(a_number)`` that checks if the number is
+    positive or negative. If ``a_number`` is positive, the function will return half the value of ``a_number``. If
+    the number is not positive, simply return ``a_number``.
 
 
+Conditional Statements
+======================
 
-.. admonition:: Activity
+* Booleans were discussed in the previous topic
 
-    Using **only** the Python features/statements **we've seen so far**, can you write a program that will divide a number in half *only if* that number is a multiple of 2?
-   
-* We need to work towards *conditional execution*
+    * We discussed them as a type
+    * How to use Boolean operators
+    * How to use comparison operators
 
+* But Booleans allow us to do a lot more than just evaluate an expression to ``True``/``False``
+* Consider ``smush`` from the above activity
+* We know how to *ask* if ``a_number`` is positive
 
-Conditional execution
-=====================
-* Now we can test if a comparison statement is ``True`` or ``False``.
-* We need a way to use that to control our program.
-* ``if`` some condition is ``True``, do something::
+    * ``a_number > 0``
+
+* The key now is to tell Python that, ``if`` some condition is ``True``, do something
+
+.. code-block:: python
+    :linenos:
+    :lines-emphasis: 11
+
+    def smush(a_number: float) -> float:
+        """
+        Returns half the value of the parameter a_number if the value is positive,
+        otherwise, return the value of a_number.
+
+        :rtype: float
+        :param a_number: Some arbitrary number.
+        :return: Half of a_number when it is positive, a_number when not positive.
+        """
+        return_value = a_number
+        if a_number > 0:
+            return_value = return_value / 2
+        return return_value
```

Comment:
Stylistic choice, you should _always_ have an `else` block.

Reply:
The only reason I didn't here was because I'm hitting `else` in the next next section.

Reply:
Fair enough. I don't think `else` would add much weight here but it will definitely add at least a little weight (imo, it makes the code simpler, but I also have non-zero experience).

Reply:
Perhaps the best way to hit this is to write `smush_version_2` that uses `else`.